File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    cnt = 0
    zero = 0
    
    for i in lottos:
        if i == 0:
            zero += 1
            
        elif i in win_nums:
            cnt += 1
    
    def rank(answer, count):
        if count == 6:
            answer += [1]
        elif count == 5:
            answer += [2]
        elif count == 4:
            answer += [3]
        elif count == 3:
            answer += [4]
        elif count == 2:
            answer += [5]
        else:
            answer += [6]
            
    rank(answer, cnt+zero)        
    rank(answer, cnt)
        
                
    
    return answer

File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        int zero = 0;
        int cnt = 0;
        
        for(int num : lottos) {
            if(num == 0) zero++;
            
            for(int win : win_nums) {
                if(num == win) {
                    cnt++;
                    break;
                }
            }
        }
        
        answer[0] = (cnt + zero == 0) ? 6 : 7 - (zero + cnt);
        answer[1] = (cnt == 0) ? 6 : 7 - cnt;
        
        return answer;
    }
}
